File: ruby/example_create_invoice.rb
# Author: Daniel Mauno Pettersson at Agigen http://agigen.se/

require './billogram_client.rb'

# Setup your credentials
api_id = "4e8eddb37cff852384e8eddb37d019"
password = 'myPassword'

# Connect to the API
billogram = BillogramClient.new( api_id: api_id, password: password)

# Create a new invoice
new_invoice = billogram.createInvoice(
    invoice_date: '2011-11-18',
    due_days: 30,
    customer_no: 1,
    delivery_type: 'Letter',
    items: {
         0 => {title: 'Cake', num: 2, price: 139}
    }
)

# Show some debug information about the newly created invoice
puts new_invoice

# Load back the same invoice from Billogram
same_invoice = billogram.getInvoice(new_invoice['invoice_no'])
# ...and show some debug information
puts same_invoice

File: php/example_create_invoice.php
<?php

include_once('AtlasClient.php');

/**
  * This code will create and send an invoice to the customer Testcompany AB.
  * The customer will also be saved and stored. The total invoice sum will be 2500 SEK.
  * @author Carl Oscar Aaro at Agigen http://agigen.se/
  */

$data = array(
    'customer_name' => 'Testcompany AB',
    'customer_org_no' => '555555-5555',
    'customer_address' => 'Teststreet 1',
    'customer_zipcode' => '10000',
    'customer_city' => 'Stockholm',
    'customer_country' => 'se',
    'invoice_date' => date('Y-m-d'),
    'due_days' => 30,
    'delivery_type' => 'letter',
    'items' => array(
        array(
            'title' => 'Technical support',
            'description' => 'Setup of example.org',
            'unit' => 'hours',
            'price' => 1000,
            'num' => 4,
            'vat' => 25,
            'discount' => 2000,
        ),
    ),
    'callback' => 'https://mywebshop.example.org/example_callback_notify.php',
    'event' => 'send',
);
$invoice = AtlasClient::createInvoice($data);
